hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -124,6 +124,10 @@ public class SCMNodeManager implements NodeManager {
   private final boolean useHostname;
   private final ConcurrentHashMap<String, Set<String>> dnsToUuidMap =
       new ConcurrentHashMap<>();
+  private final Map<String, Set<String>> ipAddressToUuidMap =
```

Comment:
I think we should only keep a single map and iterate and filter vs maintaining each way to list datanodes in memory all the time. It seems wasteful and prone to bugs on updates.

Reply:
We could have a map that holds both IP and hostnames.